handle_async.gs

```diff
@@ -5,62 +5,82 @@
-  return contentServerJsonReply(message);
+function processFunctionAsync(cmdName, args){
+  // Queue an async reply with the asyncMethod defined in globalVariables (i.e. either a time-based or form-based trigger)
+  var asyncMethod = globalVariables().ASYNC_METHOD;
+  GlobalFuncHandle[asyncMethod](cmdName, args);
 }
 
-
 /**
- * Async run functionName, and post the results to reply_url, while immediately
- * returning immediateReturnMessage.
- * @param {*} functionName
+ * Submit request to event form, to allow a delayed response
+ * @param {*} cmdName
  * @param {*} args
- * @param {*} reply_url
- * @param {*} immediateReturnMessage
  */
-function processFunctionAsync(functionName, args, reply_url, immediateReturnMessage){
-  // Queue an async reply with the asyncMethod defined in globalVariables (i.e. either a time-based or form-based trigger)
-  var asyncMethod = globalVariables().ASYNC_METHOD;
-  GlobalFuncHandle[asyncMethod](functionName, args, reply_url);
+function processAsyncWithFormTrigger(cmdName, args) {
+  // construct post request
+  var eventForm = JSON.parse(PropertiesService.getScriptProperties().getProperty('EVENT_FORM'));
+  var options = {
+    method:'post',
+    payload:{
+      [eventForm.entry_id.fctName]:cmdName,
+      [eventForm.entry_id.args]:JSON.stringify(args)
+    }
+  };
   
-  // Return immediate response to user
-  return contentServerJsonReply(immediateReturnMessage);
+  // Post request submission to form. The return value of .getContextText() does not appear to be informative of success of submission.
+  UrlFetchApp.fetch(eventForm.url, options).getContentText(); 
 }
 
 /**
- * Run functionName, and post the results to reply_url. Log the results to the
+ * Handle the submissions that originate specifically from the eventForm
+ * @param {*} values
+ */
+function handleEventFormSubmission(values){
+  // extract functionName, args and response_url
+  var [timestamp, cmdName, args_str] = values;
+  var args = JSON.parse(args_str);
+  
+  // call processAndPostResults
+  processAndPostResults(cmdName, args);
+}
+
+/**
+ * Instantiate and execute ConcreteCommand based on cmdName+args, and post the results to args.response_url. Log the results to the
  * log sheet.
- * @param {*} functionName
+ * @param {*} cmdName
  * @param {*} args
- * @param {*} reply_url
  */
-function processAndPostResults(functionName, args, reply_url){
-  var message = GlobalFuncHandle[functionName](args);
-  var return_message = postToSlack(message, reply_url);
-  var log_sheet = new LogSheetWrapper();
-  log_sheet.appendRow(
-    [new Date(), args.uniqueid, "admin",'postReply', return_message]);
+function processAndPostResults(cmdName, args){
+  try{
+    var commandWrapper = createCommandClassInstance(cmdName, args);
+    var message = commandWrapper.execute(); 
+    slackUserReply (message, args.uniqueid, args.response_url);
+  }
+  catch(errObj){
+    if (errObj instanceof TypeError || errObj instanceof ReferenceError){
```

Comment:
It's probably not necessary to this change now, but a better way of doing this would be to define our own error type, something like `HelpRequestError`, catch only this instead of having to list all the errors you don't want to catch.